From 44487fe0123a1cfeb26cebe69f52fb1d4f0f2e2f Mon Sep 17 00:00:00 2001
From: Steve Dickson <SteveD@redhat.com>
Date: Thu, 26 Jun 2014 15:33:32 -0400
Subject: [net] sunrpc: Ensure call_status() deals correctly with SOFTCONN tasks

Message-id: <1403796815-28580-27-git-send-email-steved@redhat.com>
Patchwork-id: 84868
O-Subject: [RHEL7.1 PATCH 26/29] SUNRPC: Ensure call_status() deals correctly with SOFTCONN tasks
Bugzilla: 1111170
RH-Acked-by: J. Bruce Fields <bfields@redhat.com>

From: Trond Myklebust <trond.myklebust@primarydata.com>

Signed-off-by: Trond Myklebust <trond.myklebust@primarydata.com>
Upsteam Id: 9455e3f43b017f560daf4289d0fa295a33976f2a
Bugzilla: 1111170
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/net/sunrpc/clnt.c b/net/sunrpc/clnt.c
index e74e5ed..ef740b2 100644
--- a/net/sunrpc/clnt.c
+++ b/net/sunrpc/clnt.c
@@ -2012,6 +2012,10 @@ call_status(struct rpc_task *task)
  case -EHOSTDOWN:
  case -EHOSTUNREACH:
  case -ENETUNREACH:
+  if (RPC_IS_SOFTCONN(task)) {
+   rpc_exit(task, status);
+   break;
+  }
   /*
    * Delay any retries for 3 seconds, then handle as if it
    * were a timeout.
-- 
1.7.1